test/cli_rich_container_test.go

```diff
@@ -157,74 +182,111 @@ func (suite *PouchRichContainerSuite) TestRichContainerInitdWorks(c *check.C) {
 
 	// --privileged is MUST required
 	command.PouchRun("run", "-d", "--privileged", "--rich", "--rich-mode", "sbin-init",
-		"--name", funcname, centosImage, "/usr/bin/sleep 10000").Assert(c, icmd.Success)
+		"--name", cname, centosImage, "/usr/bin/sleep 10000").Assert(c, icmd.Success)
 
-	output := command.PouchRun("inspect", funcname).Stdout()
+	output := command.PouchRun("inspect", cname).Stdout()
 	result := []types.ContainerJSON{}
 	if err := json.Unmarshal([]byte(output), &result); err != nil {
 		c.Errorf("failed to decode inspect output: %v", err)
 	}
 	c.Assert(result[0].Config.Rich, check.Equals, true)
 	c.Assert(result[0].Config.RichMode, check.Equals, "sbin-init")
 
-	c.Assert(checkPidofProcessIsOne(funcname, "/sbin/init"), check.Equals, true)
-	c.Assert(checkPPid(funcname, "sleep", "1"), check.Equals, true)
+	c.Assert(checkPidofProcess(cname, "/sbin/init"), check.Equals, true)
+	c.Assert(checkPPid(cname, "sleep", "1"), check.Equals, true)
 
 	// stop and start could work well.
-	command.PouchRun("stop", funcname).Assert(c, icmd.Success)
-	command.PouchRun("start", funcname).Assert(c, icmd.Success)
-	c.Assert(checkPidofProcessIsOne(funcname, "/sbin/init"), check.Equals, true)
-	c.Assert(checkPPid(funcname, "sleep", "1"), check.Equals, true)
+	command.PouchRun("stop", cname).Assert(c, icmd.Success)
+	command.PouchRun("start", cname).Assert(c, icmd.Success)
+	c.Assert(checkPidofProcess(cname, "/sbin/init"), check.Equals, true)
+	c.Assert(checkPPid(cname, "sleep", "1"), check.Equals, true)
 
 	// pause and unpause
-	command.PouchRun("pause", funcname).Assert(c, icmd.Success)
-	command.PouchRun("unpause", funcname).Assert(c, icmd.Success)
-	c.Assert(checkPidofProcessIsOne(funcname, "/sbin/init"), check.Equals, true)
-	c.Assert(checkPPid(funcname, "sleep", "1"), check.Equals, true)
+	command.PouchRun("pause", cname).Assert(c, icmd.Success)
+	command.PouchRun("unpause", cname).Assert(c, icmd.Success)
+	c.Assert(checkPidofProcess(cname, "/sbin/init"), check.Equals, true)
+	c.Assert(checkPPid(cname, "sleep", "1"), check.Equals, true)
 
-	command.PouchRun("rm", "-f", funcname)
+	command.PouchRun("rm", "-f", cname)
 }
 */
 
+// waitSystemdPullProcess to wait util the specified process running.
+// systemd will first pull the systemd processes, then pull the container
+// cmd process. So we need to wait the cmd process running before check it.
+func waitSystemdPullProcess(c *check.C, cname, processName string) {
+	// wait systemd pull a specified process
+	check := make(chan struct{})
+	timeout := make(chan bool, 1)
+	// set timeout to wait container started
+	go func() {
+		time.Sleep(10 * time.Second)
+		timeout <- true
```

Comment:
after send it, you can close it `close(timeout)` to avoid channel leak.

Reply:
make sense, updated